src/api/procedures/setTransferRestrictions.ts

```diff
@@ -286,14 +343,55 @@ export async function prepareStorage(
     });
   }
 
+  const op = transferRestrictionTypeToStatOpType(type, context);
+
+  const claimTypes =
+    type === TransferRestrictionType.Count || type === TransferRestrictionType.Percentage
+      ? [undefined]
+      : [ClaimType.Accredited, ClaimType.Affiliate, ClaimType.Jurisdiction];
+  const exemptionFetchers: Promise<
+    [
+      StorageKey<
+        [
+          PolymeshPrimitivesTransferComplianceTransferConditionExemptKey,
+          PolymeshPrimitivesIdentityId
+        ]
+      >,
+      bool
+    ][]
+  >[] = [];
+  claimTypes.forEach(claimType => {
+    exemptionFetchers.push(
+      statistics.transferConditionExemptEntities.entries({
+        asset: tickerKey,
+        op,
+        claimType,
+      })
+    );
+  });
+  const rawCurrentExemptions = flatten(await Promise.all(exemptionFetchers));
+  const currentExemptions: ExemptionRecords = {};
+  rawCurrentExemptions.forEach(([key]) => {
```

Comment:
```suggestion
  });
  
  const rawCurrentExemptions = flatten(await Promise.all(exemptionFetchers));
  const currentExemptions: ExemptionRecords = {};
  
  rawCurrentExemptions.forEach(([key]) => {
```

You should destructure key and args here